src/sql/src/pure.rs

```diff
@@ -687,3 +786,66 @@ async fn get_remote_csr_schema(
         confluent_wire_format: true,
     })
 }
+
+async fn compile_proto(
+    subject_name: &String,
+    ccsr_client: Client,
+) -> Result<CsrSeedCompiledEncoding, anyhow::Error> {
+    lazy_static! {
+        static ref WELL_KNOWN_REGEX: Regex = Regex::new(r#"(\.)?google\.protobuf\.\w+"#).unwrap();
+        static ref MISSING_IMPORT_ERROR: Regex =
+            Regex::new(r#"protobuf path \\"(?P<reference>.*)\\" is not found in import path"#)
+                .unwrap();
+    }
+
+    let (primary_subject, dependency_subjects) = ccsr_client.get_all_subjects(subject_name).await?;
+
+    let primary_proto_name = primary_subject.name.clone();
+    let include_dir = tempfile::tempdir()?;
+    let primary_proto_path = include_dir.path().join(&primary_proto_name);
+
+    for subject in iter::once(primary_subject).chain(dependency_subjects.into_iter()) {
+        if WELL_KNOWN_REGEX.is_match(&subject.name) {
+            continue;
+        }
+        let subject_pb = PathBuf::from(subject.name);
+        if let Some(parent) = subject_pb.parent() {
+            fs::create_dir_all(include_dir.path().join(parent))?;
+        }
+        let path = include_dir.path().join(subject_pb);
+        let bytes = strconv::parse_bytes(&subject.schema.raw)?;
+        fs::write(&path, &bytes)?;
```

Comment:
Oh, we're not get about this in this file, but this `fs::write` and the `fs::create_dir_all` should use `tokio::fs` instead so that they can call `.await`.